### PR TITLE
Add BookController

### PR DESCRIPTION
**Pull request by tarasulo**

### CODE REVIEW

File: src/main/java/library/app/spring/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Controller
+@RequestMapping("/book")
+public class BookController {
+
+    @Autowired
+    private BookService bookService;
+
+    @GetMapping("/all")
```
Comment (Sofasmile) [inline src/main/java/library/app/spring/controller/BookController.java]: what about other methods?
for example: findByTitle, add

File: src/main/java/library/app/spring/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Controller
+@RequestMapping("/book")
+public class BookController {
```
Comment (Sofasmile) [inline src/main/java/library/app/spring/controller/BookController.java]: add controller for rent

File: src/main/java/library/app/spring/controller/BookController.java (hunk 1 above)
Comment (tarasulo) [reply src/main/java/library/app/spring/controller/BookController.java]: done

File: src/main/java/library/app/spring/controller/BookController.java (hunk 2 above)
Comment (tarasulo) [reply src/main/java/library/app/spring/controller/BookController.java]: done

File: src/main/java/library/app/spring/dao/impl/UserDaoImp.java [hunk 3]
```diff
@@ -26,4 +26,12 @@ public void add(User user) {
                 .createQuery("from User", User.class);
         return query.getResultList();
     }
+
+    @Override
+    public User getById(Long id) {
+        TypedQuery<User> query = sessionFactory.getCurrentSession()
+                .createQuery("from User WHERE id=:id", User.class);
```
Comment (Sofasmile) [inline src/main/java/library/app/spring/dao/impl/UserDaoImp.java]: User user =  session.get(User.class, id);

File: src/main/java/library/app/spring/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Autowired
+    private BookService bookService;
+
+    @GetMapping("/allBooks")
```
Comment (kolyansv) [inline src/main/java/library/app/spring/controller/BookController.java]: "/all" is enough. Full path would be "/book/all". Method is in the Book controller so no need to specify "/allBooks"

File: src/main/java/library/app/spring/controller/IndexController.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package library.app.spring.controller;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+@RequestMapping("/index")
+public class IndexController {
+    @GetMapping("/")
```
Comment (kolyansv) [inline src/main/java/library/app/spring/controller/IndexController.java]: Doesn't it redundant? I mean "/"

File: src/main/java/library/app/spring/controller/RentController.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Controller
+@RequestMapping("/rent")
+public class RentController {
+    private static final Long USER_ID = 1L;
+
+    @Autowired
+    private LibraryService libraryService;
```
Comment (kolyansv) [inline src/main/java/library/app/spring/controller/RentController.java]: Best practice says to inject dependencies via constructor

File: src/main/java/library/app/spring/dao/impl/UserDaoImp.java (hunk 3 above)
Comment (tarasulo) [reply src/main/java/library/app/spring/dao/impl/UserDaoImp.java]: done

File: src/main/java/library/app/spring/controller/BookController.java (hunk 4 above)
Comment (tarasulo) [reply src/main/java/library/app/spring/controller/BookController.java]: complete

File: src/main/java/library/app/spring/controller/IndexController.java (hunk 5 above)
Comment (tarasulo) [reply src/main/java/library/app/spring/controller/IndexController.java]: I tested it. in fact, it works on both ways

**Pull request merged**